subsys/dap/Kconfig

```diff
@@ -17,6 +17,13 @@ config CMSIS_DAP_PACKET_COUNT
 	help
 	  Maximum packet buffers for request and response data.
 
+config CMSIS_DAP_PACKET_SIZE
+	int "Maximum packet size for request and response data."
```

Comment:
Either this needs to be renamed (if it refers to some protocol level packets, i.e. "transfers" in USB terminology) or completely removed (if it refers to wMaxPacketSize field in CONFIGURATION descriptor).

The endpoints are bulk, therefore the only valid options (Universal Serial Bus Specification Revision 2.0 5.8.3 Bulk Transfer Packet Size Constraints) are:
- 8, 16, 32 or 64 on Full-Speed device (or High-Speed capable device operating at Full-Speed)
- 512 on High-Speed device operating at High-Speed

Legacy USB stack (i.e. not device-next) is not really suitable for properly handling High-Speed (the problem is with High-Speed device operating at Full-Speed). The next stack is better on this.

Reply:
Would it be enough to set the default based `USB_DC_HAS_HS_SUPPORT` and add a help text?
Or do you think the risk of misconfiguration is too high?

Reply:
I don't think it should be Kconfig at all. There are already way too many Kconfigs that can give you essentially-broken-but-compiling code and this is going to be just another one.

Reply:
This Kconfig option is not exclusive for USB transport. You have to adapt it according your transport layer needs.

Reply:
Right, the idea was to potentially have DAP over BLE at some point.